src/index.js

```diff
@@ -120,14 +128,11 @@ export const rules = {
             },
           },
           // schemaJson, schemaJsonFilepath and schemaString are mutually exclusive:
-          oneOf: [{
-            required: ['schemaJson'],
+          allOf: [{
```

Comment:
something is a little _off_ about this configuration validation. my assumption is that `projectName` should be mutually exclusive with `schemaJson`, `schemaString`, and `schemaJsonFilepath`, so this configuration validation should reflect that.

additionally, we'd like to make sure that those validations only happen when those configuration properties are present, and if they're not present, it should still be valid.

there's also a section [here](https://github.com/graphcool/eslint-plugin-graphql/blob/0e2581f142915d9a93e9913d3ea5257b7c5b5434/src/index.js#L192-L201) that i think needs to match this configuration as well. can we just extract this configuration key validation and use it for all the rules?